src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -534,24 +529,18 @@ REDHAWK_PALEXPORT UInt32_BOOL REDHAWK_PALAPI PalAllocateThunksFromTemplate(HANDL
 
 REDHAWK_PALEXPORT void REDHAWK_PALAPI PalSleep(uint32_t milliseconds)
 {
-#if HAVE_CLOCK_MONOTONIC
-    timespec endTime;
-    clock_gettime(CLOCK_MONOTONIC, &endTime);
-    TimeSpecAdd(&endTime, milliseconds);
-    while (clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &endTime, NULL) == EINTR)
-    {
-    }
-#else // HAVE_CLOCK_MONOTONIC
-    timespec requested;
+    timespec requested, remaining;
```

Comment:
I would prefer keeping the TIMER_ABSTIME version here, since it is not subject to time drifting in case of interruption when the scheduler schedules out the thread and when it takes some time to schedule it back.